src/main/java/com/kolmakova/task2/builder/impl/FlowersDomBuilder.java

```diff
@@ -0,0 +1,160 @@
+package com.kolmakova.task2.builder.impl;
+
+import com.kolmakova.task2.entity.*;
+import com.kolmakova.task2.exception.ParseXmlException;
+import com.kolmakova.task2.parser.FlowerXmlAttribute;
+import com.kolmakova.task2.parser.FlowerXmlTag;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.time.YearMonth;
+import java.util.HashSet;
+import java.util.Set;
+
+public class FlowersDomBuilder extends AbstractFlowersBuilder {
+
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    private final DocumentBuilder documentBuilder;
+
+    public FlowersDomBuilder(String fileName, String schemaLocation) throws ParseXmlException {
+        super(fileName, schemaLocation);
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        try {
+            documentBuilder = factory.newDocumentBuilder();
+        } catch (ParserConfigurationException e) {
+            throw new ParseXmlException("Flowers DomBuilder not created", e);
+        }
+    }
+
+    @Override
+    protected Set<AbstractFlower> buildFlowersInternal() throws ParseXmlException {
+        Set<AbstractFlower> flowers = new HashSet<>();
+        try {
+            Document document = documentBuilder.parse(new FileInputStream(fileName));
```

Comment:
FileInputStream is not closed.
Use try with resources to automatically close FileInputStream